src/tablecloth/time/derived.clj

```diff
@@ -0,0 +1,25 @@
+(ns tablecloth.time.derived)
+
+(defn add-derived
```

Comment:
Perhaps `add-derived-metadata` or even `add-validatable-metadata` per our discussion? I have a bit of trouble understanding what derived is in reference to here. I think that's why I'm struggling with the word derived.

Reply:
I think the reason I like `add-validatable-metadata` is b/c I think it's true that we are still just adding metadata, only we are adding a more complex data structure surrounding the metadata itself (i.e. what is currently called `data`) so that we can say whether that metadata is still valid. 

Its validity is perhaps what is "derived" since it is determined by the identical-ness of the specified columns with those columns at the time we added the metadata. 

Not sure if that helps... but all I'll leave it here nonetheless :)